ppl/zqd/auth.go

```diff
@@ -0,0 +1,165 @@
+package zqd
+
+import (
+	"context"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"net/http"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	"github.com/brimsec/zq/pkg/fs"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/zap"
+)
+
+const (
+	// authTokenContextValue is the string the jwtmiddleware library uses to
+	// store the validated & parsed JWT in a request's context.
+	authTokenContextValue = "zqd-core-auth-token"
+
+	tenantIDClaim = "brim_tenant_id"
+	userIDClaim   = "brim_user_id"
+)
+
+type AuthConfig struct {
+	Enabled  bool   `env:"ZQD_AUTH_ENABLED"`
```

Comment:
I found it kind of confusing that the AuthConfig reference in core.Config didn't have `env:"ZQD"`. Should the fromenv be able to preprend children of a config var with `PARENTTAG_`?